### DISCUSSION TITLE
Bootstrapping (Server Side) Worker Threads on start

### DISCUSSION THREAD

**Discussion by jarzynka** [vercel/next.js#17087] [Help]
Hi, folks!  I'm working on an Intranet application that manages and distributes weather data.  This was originally a Node/Express application with static front end in React.  Next.js has been a God-send!  If nothing else, the elegant Routing system and dynamic front end developing has been an amazing time saver!

I am running into an "issue", though.  If, for whatever reason, the server needs to restart, I want to be able to start several child_process tasks that run "headless".  These child processes do things like autonomously fetch weather data, format it and send the data to different platforms on the Intranet.  With Express, the index.js (server) file served as a way to "bootstrap" the initialization and start up of these processes.

It doesn't seem quite as straightforward in Next.js.  I thought the _app.js file might be the "entry point" but it appears to render (often on the client side) every time a web page is loaded.  I came up with a ugly workaround of adding a useEffect to call a server API which then starts the applications with child process fork.  The problem?  If someone doesn't actually call up a page in a browser, it may not run.  If a systems engineer at the company needs to restart the server, the background tasks _may_ not start until someone visits a web page.

Although I could run these worker tasks as a completely separate node.js project (and use something like Socket.IO to communicate between my API and the worker tasks) I'm thinking there may just be something that I'm missing.  I looked at items like getInitialProps() or getServerSideProps() but these appear to run for every page request.  I'm just looking for something to run when the server starts as I was doing with Express.

If anyone has any suggestions on how I could better architect this solution, I'm all ears, too!  Thanks!!!

**Comment by hbel** [comment]
I am currently stuck at exactly the same position. I have some Persistence functions that must be run right after the application was started. I know I can accomplish that by writing a custom server, but I am pretty sure there is a much simpler option available. But sadly, I can't find anything about this in the docs.

**Comment by kevinzhuang10** [comment]
I'm running into this challenge as well. This is pretty helpful if you are willing to considering using integrated services:

https://vercel.com/docs/solutions/cron-jobs

**Comment by Amn3s1a2018** [comment]
I used custom server for something like this, and it worked quite well. 

https://nextjs.org/docs/advanced-features/custom-server

Although according to the docs "server.js doesn't go through babel or webpack", you can modify the scripts in package.json and voilä, you can use typescript and ES6 modules in your custom server.
